src/coreclr/System.Private.CoreLib/src/System/Threading/ThreadPool.CoreCLR.Windows.cs

```diff
@@ -9,48 +10,184 @@ namespace System.Threading
 {
     public static partial class ThreadPool
     {
+        internal static readonly bool UseWindowsThreadPool =
```

Comment:
This should be integrated with trimming to trim the threadpool implementation that the app does not need.